### PR TITLE
feat(kcardcatalog): adds fetcher and integrates backend pagination

### PR DESCRIPTION
**Pull request by aanchalm01**
### Summary
Adds the ability to pass a fetcher function into a new fetcher prop which will enable backend pagination for the card catalog.

#### Changes made:
* Many new props added related to fetcher/pagination see the docs
* Dependencies added:` composition-api` and `swrv`

### PR Checklist
- [ ] Does not introduce dependencies
- [x] **Functional:** all changes do not break existing APIs and if so, bump major version.
- [x] **Tests pass:** check the output of yarn test packages/<Kongponent>
- [x] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
- [x] **Framework style:** abides by the essential rules in Vue's style guide
- [x] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs), or leftover comments
- [x] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

### CODE REVIEW

File: docs/components/card-catalog.md [hunk 1]
```diff
@@ -2,39 +2,19 @@
...
-<KCardCatalog :items="getItems(5)" />
 ```vue
 <KCardCatalog :items="items" />
 ```
 
+Pass a fetcher function to build a slot-able card catalog.
+
+```vue
+<KCardCatalog 
+   :fetcher="fetcher" />
```
Comment (adamdehaven) [inline docs/components/card-catalog.md:16]: Can you move this up on the same line as the component name?

**Review by adamdehaven — CHANGES_REQUESTED** [review]
Looks good to me -- just a couple small changes

File: packages/KCardCatalog/KCardCatalog.vue [hunk 2]
```diff
@@ -269,14 +280,125 @@ export default {
...
+      type: Number,
+      default: 1
+    },
+    /**
+     * A prop to pass in an array of page sizes used by the pagination component
+     */
+    paginationPageSizes: {
+      type: Array,
+      default: () => ([15, 25, 50, 75, 100])
+    },
```
[the remark this answers is not in the document]
Comment (aanchalm01) [reply packages/KCardCatalog/KCardCatalog.vue]: Done!

File: docs/components/card-catalog.md (hunk 1 above)
Comment (aanchalm01) [reply docs/components/card-catalog.md]: Done!

**Review by adamdehaven — APPROVED** [review]

**Pull request merged**